src/parser/core/modules/EventEmitter.ts

```diff
@@ -16,6 +16,11 @@ type BoundListener<ET extends EventType, E extends MappedEvent<ET>> = {
   listener: EventListener<ET, E>;
 };
 
+interface FabricatedEvent extends Omit<Event<any>, 'timestamp'> {
+  timestamp?: number | undefined;
+  [prop: string]: any;
+}
```

Comment:
I'm unclear why you're doing this - which FabricatedEvent doesn't have a timestamp?

Reply:
This is the error I get before changing the type:
<img width="1139" alt="Screen Shot 2021-06-25 at 17 54 18" src="https://user-images.githubusercontent.com/24693913/123443230-67735600-d5de-11eb-8c66-b4e319a80298.png">

Same thing happens for current:
<img width="1179" alt="Screen Shot 2021-06-25 at 17 56 47" src="https://user-images.githubusercontent.com/24693913/123443676-c769fc80-d5de-11eb-9bea-528da649df77.png">

This seems ok though:
<img width="565" alt="Screen Shot 2021-06-25 at 18 00 27" src="https://user-images.githubusercontent.com/24693913/123444134-3ba4a000-d5df-11eb-9575-3c196af46ce9.png">

I've gone over some other classes and they seemed to implemented it with variables like so:

```ts
    const evt: AtonementAnalyzerEvent = {
      timestamp: spiritShellEvent.timestamp,
      targetID: spiritShellEvent.targetID,
      healEvent: spiritShellEvent,
      damageEvent: this.atonementSource,
      sourceID: spiritShellEvent.sourceEvent.sourceID,
      provenance: SourceProvenance.SpiritShell,
      type: AtonementAnalyzer.atonementEventFilter.eventType,
      buffDetails,
      expirationDelta: buffDetails.baseExpirationTimestamp - spiritShellEvent.timestamp,
    };

    this.eventEmitter.fabricateEvent(evt);
```

I'm not that fluent with TS, could this be the problem here?:
https://stackoverflow.com/questions/61698807/interesting-behaviour-object-literal-may-only-specify-known-properties

---

If I remove Omit and just extend Event, as far as TS concerns, timestamp should never be undefined therefor doesn't require a default assignment.
<img width="544" alt="Screen Shot 2021-06-25 at 18 07 43" src="https://user-images.githubusercontent.com/24693913/123445105-3eec5b80-d5e0-11eb-80d5-b73d6473e3c9.png">

These are the problems I run into. So 2 more possible solutions I see here:
- Just implement it as an event variable and revert changes in EventEmitter.ts.
- Remove default assignment as it should never be undefined.

Reply:
I still haven't had time to look really closely at what's happening, but it sounds like wires got crossed somewhere. I'd like you to revert the changes to EventEmitter - an Event, even a fabricated one, should always have a timestamp. If you're sure the timestamp you're using will never be undefined, I'd prefer you make whatever assertions are needed when the Event is created instead of altering Event's signature.

Reply:
Okay, I'll revert the EventEmitter changes and go with an event variable so that TS doesn't complain.

Reply:
Done. I reverted the emitter changes.

Reply:
Looks good. Odd that assigning it to a separate var was all it took to calm the typechecker...